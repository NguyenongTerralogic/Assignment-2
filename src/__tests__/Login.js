import SignIn from '../screens/sign-in';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({
    adapter: new Adapter(),
});

describe('test for SignIn component', () => {
    const wrapper = shallow(<SignIn />);
    it('it should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('test for SignIn component', () => {
    const wrapper = shallow(<SignIn />);
    const instance = wrapper.instance();
    const loginClick = jest.spyOn(instance);
    wrapper.update();
    instance.forceUpdate();
    it('it should render with check func', () => {
        const email = wrapper.find(".email-login");
        const password = wrapper.find(".password-login");
        const loginClick = wrapper.find(".login-button");
        loginClick.simulate("click");
        expect(email !== "").toBeTruthy();
        expect(password !== "").toBeTruthy();
    });
});