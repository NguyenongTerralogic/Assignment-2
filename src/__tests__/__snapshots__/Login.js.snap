// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for SignIn component it should render 1`] = `ShallowWrapper {}`;
